lib/whois/record/parser/whois.tld.ee.rb

```diff
@@ -20,9 +18,111 @@ class Parser
       # @see Whois::Record::Parser::Example
       #   The Example parser for the list of all available methods.
       #
-      class WhoisTldEe < BaseWhoisd
-      end
+      class WhoisTldEe < Base
+        include Scanners::Scannable
+        class_attribute :status_mapping
+
+        self.status_mapping = {
+          'ok (paid and in zone)' => :registered,
+          'expired' => :expired
+        }
+
+        self.scanner = Scanners::WhoisTldEe
+
+        property_supported :available? do
+          !!node('status:available')
+        end
+
+        property_supported :registered? do
+          !available?
+        end
+
+        property_supported :domain do
+          if (content_for_scanner =~ /^Domain:\nname:\s+(.+)\n/)
+            $1.to_s.strip.downcase
+          end
+        end
+
+        property_not_supported :domain_id
+
+        property_supported :status do
+          if content_for_scanner =~ /status:\s+(.+?)\n/
+            self.class.status_mapping[$1] || $1
```

Comment:
As the parser is not generic but specific for this TLD, rather than setting a class-level hash I'd rather use a `case` switch. Here's an example.
https://github.com/weppos/whois/blob/master/lib/whois/record/parser/whois.cira.ca.rb#L41-L64